rs/execution_environment/tests/storage_reservation.rs

```diff
@@ -0,0 +1,132 @@
+use ic_config::execution_environment::Config as ExecutionConfig;
+use ic_config::subnet_config::SubnetConfig;
+use ic_management_canister_types::{self as ic00, CanisterInstallMode, EmptyBlob, Payload};
+use ic_registry_subnet_type::SubnetType;
+use ic_state_machine_tests::{StateMachine, StateMachineBuilder, StateMachineConfig};
+use ic_test_utilities::universal_canister::{call_args, wasm, UNIVERSAL_CANISTER_WASM};
+use ic_types::{CanisterId, Cycles, NumBytes};
+use more_asserts::{assert_gt, assert_lt};
+
+const T: u128 = 1_000_000_000_000;
+
+const KIB: u64 = 1024;
+const MIB: u64 = 1024 * KIB;
+const GIB: u64 = 1024 * MIB;
+const SUBNET_MEMORY_THRESHOLD: u64 = 10_253 * MIB;
+const SUBNET_MEMORY_CAPACITY: u64 = 20 * GIB;
+
+const PAGE_SIZE: usize = 64 * KIB as usize;
+
+fn setup(subnet_memory_threshold: u64, subnet_memory_capacity: u64) -> (StateMachine, CanisterId) {
+    let subnet_type = SubnetType::Application;
+    let execution_config = ExecutionConfig {
+        subnet_memory_threshold: NumBytes::new(subnet_memory_threshold),
+        subnet_memory_capacity: NumBytes::new(subnet_memory_capacity),
+        ..Default::default()
+    };
+    let config = StateMachineConfig::new(SubnetConfig::new(subnet_type), execution_config);
+    let env = StateMachineBuilder::new()
+        .with_config(Some(config))
+        .with_subnet_type(subnet_type)
+        .with_checkpoints_enabled(false)
+        .build();
+    let canister_id = env.create_canister_with_cycles(None, Cycles::from(100 * T), None);
+
+    env.install_wasm_in_mode(
+        canister_id,
+        CanisterInstallMode::Install,
+        UNIVERSAL_CANISTER_WASM.to_vec(),
+        vec![],
+    )
+    .unwrap();
+
+    (env, canister_id)
+}
+
+#[test]
+fn test_storage_reservation_not_triggered() {
```

Comment:
Perhaps add a comment on why we expect that the reservation mechanism is not triggered?

Actually, this applies to all tests I guess. Maybe we have some comment that explains kinda the setup. We have a specific threshold and capacity, we expect the mechanism to trigger whenever you try to grow your memory above the threshold but not when you're below.

Reply:
I added high level comments to each test.